### ISSUE TITLE
I still experience the same issue. It there is a difference between the three places where the result is shown. rendered dice: 20. Top-right: 20. Text in obsidian, the only one that stays, btw, 3.

### ISSUE THREAD

**Issue by sigrunixia** [javalent/dice-roller#196]
I still experience the same issue. It there is a difference between the three places where the result is shown. rendered dice: 20. Top-right: 20. Text in obsidian, the only one that stays, btw, 3. 
But only with the first throw of a dice-set. 

I can reproduce by adding these lines:

`dice: 1d6`    
`dice: 3d20k` 

Click upper one, should show difference between rendered and what is shown in the text-document. switch to the lower one, same. It is not happening though, when you use the same dice twice or more times. It's only the first time it is been used.

_Originally posted by @ImmoNilsEckelt in https://github.com/javalent/dice-roller/issues/152#issuecomment-1517737895_

**Comment by sigrunixia** [comment]
Creating new issue as @ImmoNilsEckelt is still continuing to have problems (and we're bumping this to newer versions of dice roller).

**Comment by sigrunixia** [comment]
## Bugtesting
Obsidian Version: Insiders 1.2.5
Installer Version: 1.1.9
Dice Roller Version: 8.14.2

Plugins Enabled: Dice Roller

### Entering Directly into a Note in Live preview
![dice](https://user-images.githubusercontent.com/68425372/233743369-02a09730-0386-41c6-85b0-e79f3d0afc09.gif)

### Going from a different note then clicking in Live Preview
![dice-2](https://user-images.githubusercontent.com/68425372/233743486-d41a4468-f1b5-447b-b26b-045ed79406d2.gif)

### Going from a different note then clicking in reading mode
![dice-3](https://user-images.githubusercontent.com/68425372/233743632-5d2efcb6-b358-4fdb-b06b-0ebf76ef5aee.gif)

### Opening Vault
![dice-4](https://user-images.githubusercontent.com/68425372/233743739-fadb1e8a-1f6e-42d5-96b9-ef129fd9ba70.gif)


In my testing, the inline renders matched the rolls and the notification results. We may need to look at something deeper here.

**Comment by PrestaJay** [comment]
Hi, I just installed Dice Roller for the first time and I am also experiencing this issue. Below are my version details:

Obsidian Version: 1.1.16
Installer Version: 1.1.16
Dice Roller Version: 8.14.5

I am not seeing any errors in the Developer Console when I attempt this.

As with the original report, the rendered dice match the little black box that appears in the top-right, however the value shown inline in the document is different. I've attached a screenshot with the formatting set to include the results inline.

This is happening the first time I click on an inline button to roll, but subsequent clicks of the same button will not cause the issue. However, clicking a different dice button will cause the issue.

Please let me know any other information I can provide to help troubleshoot.

![Dice Roller #196 Test](https://user-images.githubusercontent.com/62760870/235380932-fcd59021-55c2-46bb-9137-fc4b5edc35cc.jpg)

**Comment by sigrunixia** [comment]
Let's look at other factors as well... 

@PrestaJay Can you include what your system language is, as well as keyboard layout?

**Comment by PrestaJay** [comment]
My system language is English (United States) and my keyboard layout is also U.S.

If it's helpful, I'm able to reproduce it on both MacOS 13 and Windows 10.

**Comment by ImmoNilsEckelt** [comment]
Just checked, still experiencing this issue. 

Obsidian: Version 1.1.16 (Installer 0.15.8)
Dice Roller Version: 8.14.5

German System Language
Keyboard Layout is US

Thanks for continue looking into this issue.

**Comment by PrestaJay** [comment]
Is there any other information I can provide to help with diagnosing or isolating?

**Comment by sigrunixia** [comment]
> Is there any other information I can provide to help with diagnosing or isolating?

Not at all. :) Summer is usually slow for Javalent, but this year it's been super busy. This is bunched to be worked with a major dice roller rewrite.
